tests/conftest.py

```diff
@@ -204,3 +205,99 @@ def _R(expr):
         assert len(base) == 1
         base = base.pop()
     return EvalDerivative(*expr.args, base=base)
+
+
+# Utilities for testing tree structure
+
+
+def assert_structure(operator, exp_trees=None, exp_iters=None):
+    """
+    Utility function that helps to check loop structure of IETs. Retrieves trees from an
+    Operator and check that blocking structure is as expected. Trees and Iterations are
+    returned for further use in tests.
+
+    Example:
+
+    To check the following structure:
+
+    .. code-block:: python
+
+        for time
+            for x
+                for y
+            for f
+                for y
+
+    we call:
+
+    `trees, iters = assert_structure(op, ['t,x,y', 't,f,y'], 't,x,y,f,y')`
+    """
+
+    mapper = {'time': 't'}
+    trees = retrieve_iteration_tree(operator)
+    iters = FindNodes(Iteration).visit(operator)
+
+    if exp_trees is not None:
+        exp_trees = [i.replace(',', '') for i in exp_trees]  # 't,x,y' -> 'txy'
+        tree_struc = (["".join(mapper.get(i.dim.name, i.dim.name) for i in j)
+                       for j in trees])
+        assert tree_struc == exp_trees
+
+    if exp_iters is not None:
+        exp_iters = exp_iters.replace(',', '')  # 't,x,y' -> 'txy'
+        iter_struc = "".join(mapper.get(i.dim.name, i.dim.name) for i in iters)
+        assert iter_struc == exp_iters
+
+    return trees, iters
+
+
+def get_blocked_nests(operator, exp_nests):
+    """
+    Utility function that helps to check existence of blocked nests. The structure of the
+    operator is not used. Trees are returned for further use in tests.
+
+    Example:
+
+    For the following structure:
+
+    .. code-block:: python
+
+        for time
+            for x0_blk0
+                for x
+            for x1_blk0
+                for x
+
+    we call:
+
+    `trees, bns = get_blocked_nests(op, {'x0_blk0', 'x1_blk0'})`
+
+    to assert the existence of 'x0_blk0', 'x1_blk0' and then the function returns the
```

Comment:
instead of (or besides) "and then the function returns", keep on with the example? show what is actually returned?